proof/verifier.go

```diff
@@ -304,6 +308,12 @@ func (p *Proof) verifyAssetStateTransition(ctx context.Context,
 		splitAssets = append(splitAssets, splitAsset)
 	}
 
+	chainLookup, err := vCtx.ChainLookupGen.GenProofChainLookup(p)
```

Comment:
We should actually pass in the `chainLookup` from `Verify`. Otherwise the caching still doesn't work.